File: php/utility/PrepareBody.php
<?php
declare(strict_types=1);

// BluefinDecryptx SDK utility: prepare_body

class BluefinDecryptxPrepareBody
{
    public static function call(BluefinDecryptxContext $ctx): mixed
    {
        if ($ctx->op->input === 'data') {
            return ($ctx->utility->transform_request)($ctx);
        }
        return null;
    }
}

File: java/utility/PrepareBody.java
package voxgig.bluefindecryptxsdk.utility;

import voxgig.bluefindecryptxsdk.core.Context;

final class PrepareBody {

  private PrepareBody() {}

  static Object prepareBody(Context ctx) {
    if ("data".equals(ctx.op.input)) {
      return ctx.utility.transformRequest.apply(ctx);
    }

    return null;
  }
}
